transcom/move.mil
# Config with structure for dynamic SME page

This PR addresses Pivotal tickets [#161924549 - SME Document Content Type](https://www.pivotaltracker.com/story/show/161924549), [#161923738 - new SME page architecture](https://www.pivotaltracker.com/story/show/161923738), and in part [#161925392 - new SME archive view/page](https://www.pivotaltracker.com/story/show/161925392). Further changes will likely be needed (and theming definitely will), but this gives us a solid starting point for team review and iteration.

## Checklist

I have…

      against my changes

## Summary of Changes

This pull request…

- Creates a new "SME Document" content type. Includes the fields indicated in the corresponding ticket - Title, ID, Updated Date, Archive/Expiration Date, and the File itself. Adds an "SME Section" taxonomy reference field.
- Customizes the form and view display modes to match structural/UI needs.
- Adds a template for the Teaser display mode that uses the Title as the text for the file's download link and adds file size accessed in Twig through the File field's metadata.
- Creates a new "SME Section" Taxonomy vocabulary, where terms corresponding to each section can be added as content ([see Pivotal for details](https://www.pivotaltracker.com/story/show/161923738))
- Creates a new "SME Section" view, and a bunch of block displays containing the content of SME Documents in Teaser view mode, each display filtered by a corresponding SME Section taxonomy term.
- Places the various blocks corresponding with the view displays in the Content section of page layout, in the order matching ticket instructions and each restricted to "/sme-page".
- In the new view, adds Archive versions of each block display AND a single Archive page display that does not include taxonomy filters. We will want to use one or the other, not both.
- Adds a new "SME Sections" menu meant to contain links to each section's id (already being generated through existing code), places it in the left sidebar and restricts it to "/sme-page". If we build the Archive page similarly, we can also use it there.

## Testing

To verify the changes proposed in this pull request…
_(tentative, this is an educated guess)_ 

1. Pull code and import config. No need to rebuild theme assets at this time.
1. Create a blank Basic page with whatever Title you want and set the custom URL to "/sme-page" (we'll change it to /sme later, once we're ready to replace the existing one).
1. Create the specified terms ("Personal Property", "Advisories", etc) within the SME Section taxonomy vocabulary
1. Add sidebar menu items, linking to each section block's id, as outlined in the section above.
1. Upload one or more items and their corresponding data as SME Document content. Ideally, do enough to verify multiple items appear in the same section and that items in different sections appear properly. 

## Screenshots

New Page on my local with some content: 
<img width="1009" alt="dynamic sme page - local prelim" src="https://user-images.githubusercontent.com/29130580/49096144-ebeb7580-f237-11e8-904a-b0382f533965.png">
